arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#include "SupportInfoBuilder.h"
+
+#include "Agency/AsyncAgencyComm.h"
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/files.h"
+#include "Basics/NumberOfCores.h"
+#include "Basics/PhysicalMemory.h"
+#include "Basics/process-utils.h"
+#include "Basics/StaticStrings.h"
+#include "Basics/StringUtils.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterInfo.h"
+#include "Cluster/ServerState.h"
+#include "GeneralServer/AuthenticationFeature.h"
+#include "Indexes/Index.h"
+#include "Metrics/MetricsFeature.h"
+#include "Network/Methods.h"
+#include "Network/NetworkFeature.h"
+#include "Replication/ReplicationFeature.h"
+#include "Rest/Version.h"
+#include "RestServer/CpuUsageFeature.h"
+#include "RestServer/DatabaseFeature.h"
+#include "RestServer/EnvironmentFeature.h"
+#include "RestServer/ServerIdFeature.h"
+#include "Statistics/ServerStatistics.h"
+#include "StorageEngine/EngineSelectorFeature.h"
+#include "StorageEngine/StorageEngine.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/DatabaseGuard.h"
+#include "Utils/SingleCollectionTransaction.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/LogicalView.h"
+#include "VocBase/Methods/Collections.h"
+#include "VocBase/Methods/Databases.h"
+#include "VocBase/Methods/Indexes.h"
+
+#include <velocypack/Builder.h>
+#include <absl/strings/str_replace.h>
+
+#include "Logger/LogMacros.h"
+
+using namespace arangodb;
+using namespace arangodb::rest;
+using namespace std::literals;
+
+namespace {
+network::Headers buildHeaders() {
+  auto auth = AuthenticationFeature::instance();
+
+  network::Headers headers;
+  if (auth != nullptr && auth->isActive()) {
+    headers.try_emplace(StaticStrings::Authorization,
+                        "bearer " + auth->tokenCache().jwtToken());
+  }
+  return headers;
+}
+}  // namespace
+
+void SupportInfoBuilder::addDatabaseInfo(VPackBuilder& result,
+                                         VPackSlice infoSlice,
+                                         ArangodServer& server) {
+  DatabaseFeature& dbFeature = server.getFeature<DatabaseFeature>();
+
+  std::vector<std::string> databases = methods::Databases::list(server, "");
+
+  containers::FlatHashMap<std::string_view, uint32_t> dbViews;
+  for (auto const& database : databases) {
+    auto vocbase = dbFeature.lookupDatabase(database);
+
+    LogicalView::enumerate(*vocbase,
```

Comment:
It is not safe to use the return value of `dbFeature.lookupDatabase(...)` unconditionally. The API returns a raw pointer, and it can return a nullptr!
Instead, please use `dbFeature.useDatabase(...)`, which returns a VocbasePtr RAII object. You still need to check whether it is a nullptr though.